### DISCUSSION TITLE
Mirror Discussions

### DISCUSSION THREAD

**Discussion by CSSFrancis** [hyperspy/hyperspy#3391] [General]
I really like the dicussions as a way to ask questions. I think it brings people to where the developers are and it is easy to transform issues to discussions and vice versa.

The problem right now is the breadth of packages hyperspy, exspy, rosettasciio, pyxem etc.

Many of the questions could be asked in different packages and sometimes it might be hard to know.  

Proposed solution:

First it would be nice to tag each discussion. This is pretty easy from a developer standpoint. We can organize questions by affected packages etc.

The second part would be to automatically mirror the questions downstream in pyxem rosettasciio etc. 

For example someone asks a question about loading files in hyperspy --> that gets tagged as `hyperspy` and `rosettasciio`. That discussion is now linked between the two repos with it showing up both places.

**Comment by CSSFrancis** [comment]
Relevant:

https://github.com/marketplace/actions/github-issue-copier

**Comment by ericpre** [comment]
As you said, there are indeed discussions for which it is not obvious at first where it should be asked. However, I am not sure this is big issue - this is not ideal, using GitHub Discussion is most likely the best that we have, see ##3374.

Setting up a GitHub workflow to create issue will help increasing visibility, but there is a risk that the discussion take place in both places, which can be very inconvenient/counterproductive. Being able to move discussion to other repository works very well for that purposes but this works only for repository within the same GitHub organisation, and will not work for pyxem and lumispy for example.

Maybe we need to write down/document in relevant places that:
- discussions on topics that are generic (not domain specific) should take place in the hyperspy repository: https://github.com/hyperspy/hyperspy/discussions
- other discussions go the corresponding repository:
  - EELS, EDS to exspy: https://github.com/hyperspy/exspy/discussions
  - scanning diffraction to pyxem: https://github.com/pyxem/pyxem/discussions
  - reading / writing files to rosettasciio; https://github.com/hyperspy/rosettasciio/discussions
- some practicalities:
  - Users can opt-in to follow the discussion only
  - discussion can be moved from one repository to another within the same organisation (by a member of this organisation)
  - functionalities to link to code and issue or pull request, which help for discoverability and find related discussions

Would that be enough?

**Reply by CSSFrancis** [reply]
@ericpre Yea you are probably right :) 

It still might be good to copy the title and link the thread in some cases. You can always link the thread and then say discussion is following the link and then lock the thread.

For example there is a question in the pyxem discussion about saving as a zipped zspy which I think would be good both there and in rosettasciio. 

Maybe this is just solved by simply having a template for a discussion post which direct people where to go :)
